src/TransactionsBundle/Service/MatchService.php

```diff
@@ -11,113 +11,135 @@
+  protected $entityManager;
+
+  public function __construct(
+    TransactionsRepository $transactionsRepository,
+    CategoriesRepository $categoryRepository,
+    EntityManager $entityManager
+  ) {
+    $this->transactionsRepository = $transactionsRepository;
+    $this->categoryRepository = $categoryRepository;
+    $this->entityManager = $entityManager;
+  }
+
+  public function match($match, $transactions, $category)
+  {
+    $results = array();
+
+    $matchDescription = $this->cleanUp($match->getDescription());
+
+    foreach ($transactions as $item) {
+      $score   = 0;
+      $special = 0;
+      $itemDescription = $this->cleanUp($item->getDescription());
+
+      foreach ($itemDescription as $item1) {
+        if (in_array($item1, $matchDescription)) {
+          $score += 1;
+        }
+      }
 
-    public function match($match, $transactions, $category)
-    {
-        $results = array();
+      $matchPercent = round((($score*100)/(count($itemDescription))),0);
 
-        $matchDescription = $this->cleanUp($match->getDescription());
+      $type = $this->transactionsRepository->findById($category);
+      $type = $this->categoryRepository->findById($category);
 
-        foreach ($transactions as $item) {
-            $score   = 0;
-            $special = 0;
-            $itemDescription = $this->cleanUp($item->getDescription());
+      if ($matchPercent >= 100 && $item->getCategories() == NULL) {
+        $item->setCategories($type[0]);
+        $item->setMatchPercentage($matchPercent);
 
-            foreach ($itemDescription as $item1) {
-                if (in_array($item1, $matchDescription)) {
-                    $score += 1;
-                }
-            }
+        $this->entityManager->persist($item);
+        $this->entityManager->flush();
 
-            $matchPercent = round((($score*100)/(count($itemDescription))),0);
+        $results[$item->getId()] = $item;
 
-            $type = $this->transactionsRepository->findById($category);
-            $type = $this->categoryRepository->findById($category);
+        $score = 0;
+        $special = 0;
 
-            if ($matchPercent >= 100 && $item->getCategories() == NULL) {
-                $item->setCategories($type[0]);
-                $item->setMatchPercentage($matchPercent);
+        continue;
+      }
+    }
+  }
+
+  public function cleanUp($description)
+  {
+    $description = preg_replace("/\w{3,}\s+\w+\:[A-Z0-9]+\s+[0-9]{2,}.[0-9]{2,}.[0-9]{2,}\/[0-9]{2,}\.[0-9]{2,}\s/",'',$description);
+    $description = preg_replace("/,\w{4,}/",'',$description);
+    $description = preg_replace("/SEPA Incasso algemeen doorlopend Incassant:/",'',$description);
+    $description = preg_replace('/(\d{1,2}[.\/])+\d{1,2}/','',$description);
+    $description = preg_replace("/\d\d-\d\d-\d\d\d\d/", "", $description);
+    $description = preg_replace('/\w+:[A-Z0-9]+/','',$description);
+    $description = preg_replace("/,PAS572/",'',$description);
```

Comment:
Not very generic - maybe put PAS572 as a variable in the parameters file or match it by `,PAS[0-9]{3}`